--- xemacs-21.5.31/src/Makefile.in.in.orig	2011-08-25 23:06:28.371099662 -0400
+++ xemacs-21.5.31/src/Makefile.in.in	2011-08-25 23:06:38.535086804 -0400
@@ -904,7 +904,7 @@
 check-temacs:
 	$(TEMACS_BATCH) $(run_temacs_args) $(batch_test_emacs)
 check-features: all
-	$(XEMACS_BATCH) -l check-features.el
+	$(XEMACS_BATCH) -l ${blddir}/lisp/check-features.el
 
 ###################### Debugging targets
 
