--- xemacs-21.5.34/lisp/find-paths.el.orig	2013-12-11 22:18:22.003073910 -0500
+++ xemacs-21.5.34/lisp/find-paths.el	2013-12-11 22:19:59.667076525 -0500
@@ -226,7 +226,7 @@
 				  roots
 				  (file-name-as-directory
 				   (paths-construct-path (list
-							  (if arch-dependent-p "lib" "share")
+							  (if arch-dependent-p "lib64" "share")
 							  emacs-program-name)))
 				  bases
 				  envvar default))
@@ -276,7 +276,7 @@
 				  roots
 				  (file-name-as-directory
 				   (paths-construct-path
-				    (list (if arch-dependent-p "lib" "share")
+				    (list (if arch-dependent-p "lib64" "share")
 					  (construct-emacs-version-name))))
 				  bases
 				  envvar default))
--- xemacs-21.5.34/lisp/setup-paths.el.orig	2013-12-11 22:29:09.908091260 -0500
+++ xemacs-21.5.34/lisp/setup-paths.el	2013-12-11 22:29:21.274091565 -0500
@@ -75,7 +75,7 @@
   (or
    ;; installed
    (paths-file-readable-directory-p (paths-construct-path (list directory
-								"lib"
+								"lib64"
 								(construct-emacs-version-name))))
    ;; in-place or windows-nt.  windows-nt equivalent of --srcdir is
    ;; BUILD_DIR in config.inc, and has no lisp/ or etc/ since symlinks
