--- xemacs-21.5.34/src/orig-select-x.c	2017-05-22 11:57:39.608566802 -0400
+++ xemacs-21.5.34/src/select-x.c	2017-05-22 12:00:06.361819319 -0400
@@ -154,6 +154,7 @@
   Display *display = DEVICE_X_DISPLAY (d);
 
   if (! atom) return Qnil;
+  if (atom == (Atom)-1) return Qnil;
   if (atom == XA_PRIMARY)	return QPRIMARY;
   if (atom == XA_SECONDARY)	return QSECONDARY;
   if (atom == XA_STRING)	return QSTRING;
